src/twisted/trial/_dist/worker.py

```diff
@@ -283,32 +304,47 @@ class LocalWorker(ProcessProtocol):
 
     @ivar _logDirectory: The directory where logs will reside.
 
-    @ivar _logFile: The name of the main log file for tests output.
+    @ivar _logFile: The main log file for tests output.
     """
 
-    def __init__(self, ampProtocol, logDirectory, logFile):
+    def __init__(
+        self,
+        ampProtocol: LocalWorkerAMP,
+        logDirectory: FilePath,
+        logFile: TextIO,
+    ):
         self._ampProtocol = ampProtocol
         self._logDirectory = logDirectory
         self._logFile = logFile
-        self.endDeferred = Deferred()
+        self.endDeferred: Deferred = Deferred()
+
+    async def exit(self) -> None:
+        """
+        Cause the worker process to exit.
+        """
+        if self.transport is None:
+            raise NotRunning()
```

Comment:
I am not sure why there is no coverage for this.
Is it expected not to have coverage?

I guess that this is just some corner care for which writing a test is not that easy, and is not worth spending the extra effort to write a test for it.


But maybe it is worth to add a nocover pragma together with a comment.

Reply:
I added a couple tests and fixed a bug they revealed. :)